### DISCUSSION TITLE
IRIS instance upgrade

### DISCUSSION THREAD

**Discussion by isc-rsingh** [intersystems/ipm#324] [Ideas]
- On instance upgrade, there should be NO change to IPM version.
- Utilities should be provided (probably via IPM shell commands) to easily install/upgrade IPM across multiple namespaces (Optionally: let IPM check for upgrade when entering the IPM shell and notify user. Provide utility to disable the notice.)

**Comment by daimor** [comment]
There is still no agreement on how IPM out of the box is supposed to be looks like
At the moment, it looks like it will be just XML file

**Reply by bg-ssh** [reply]
Again I would prefer to treat ZPM similar to Ensemble from distribution viewpoint, and replace the ZPM version during upgrade. People then can install whatever version they want if they're fancy. Otherwise we'll end up with stale insecure ZPM everywhere.
